posthog/settings/dynamic_settings.py

```diff
@@ -27,7 +28,7 @@
         str,
     ),
     "PERSON_ON_EVENTS_ENABLED": (
-        get_from_env("PERSON_ON_EVENTS_ENABLED", False, type_cast=str_to_bool),
+        get_from_env("PERSON_ON_EVENTS_ENABLED", not TEST and not E2E_TESTING, type_cast=str_to_bool),
```

Comment:
I was thinking that our test code should probably switch it's default assumption & have explicit calls for running person-on-events disabled tests while we still have an option to switch the flag off. But this is a lot faster and we're blocking the release, so 👍

Reply:
Updating tests at minimum requires:
- Updating all snapshots
- Removing a now-unneeded ci workflow
- Updating data seeding logic for cypress

I'm probably missing other aspects as well, but this is pretty massive on its own.